src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,206 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level; _⊔_)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties          as Properties
+import Function.Definitions
+import Relation.Binary.Reasoning.Setoid   as Reasoning
+import Algebra.Module.Morphism.Structures as MorphismStructures
+
+open import Agda.Builtin.Sigma
+open import Axiom.DoubleNegationElimination
+open import Data.List
+open import Data.Product
+  using (Σ; _,_; ∃; Σ-syntax; ∃-syntax; _×_)
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+open Module modA
+  using () renaming
+  ( Carrierᴹ       to A
+  ; _+ᴹ_           to _+ᴬ_
+  ; _*ₗ_           to _·ᴬ_
+  ; -ᴹ_            to -ᴬ_
+  ; _≈ᴹ_           to _≈ᴬ_
+  ; 0ᴹ             to 0ᴬ
+  ; +ᴹ-comm        to +ᴬ-comm
+  ; +ᴹ-congˡ       to +ᴬ-congˡ
+  ; +ᴹ-congʳ       to +ᴬ-congʳ
+  ; *ₗ-zeroˡ       to ·ᴬ-zeroˡ
+  ; -ᴹ‿cong        to -ᴬ‿cong
+  ; -ᴹ‿inverseʳ    to -ᴬ‿inverseʳ
+  ; -ᴹ‿involutive  to -ᴬ‿involutive
+  ; uniqueʳ‿-ᴹ     to uniqueʳ‿-ᴬ
+  ; ≈ᴹ-setoid      to ≈ᴬ-setoid
+  ; ≈ᴹ-sym         to symᴬ
+  ; leftSemimodule to leftSemimoduleᴬ
+  )
+open Properties leftSemimoduleᴬ
+  using () renaming
+  ( non-zeroʳ to non-zeroʳᴬ
+  ; non-zeroˡ to non-zeroˡᴬ
+  )
+open Module modB
+  using () renaming
+  ( Carrierᴹ       to B
+  ; _+ᴹ_           to _+ᴮ_
+  ; _*ₗ_           to _·ᴮ_
+  ; -ᴹ_            to -ᴮ_
+  ; _≈ᴹ_           to _≈ᴮ_
+  ; 0ᴹ             to 0ᴮ
+  ; +ᴹ-comm        to +ᴮ-comm
+  ; +ᴹ-congˡ       to +ᴮ-congˡ
+  ; +ᴹ-congʳ       to +ᴮ-congʳ
+  ; *ₗ-zeroˡ       to ·ᴮ-zeroˡ
+  ; -ᴹ‿cong        to -ᴮ‿cong
+  ; -ᴹ‿inverseʳ    to -ᴮ‿inverseʳ
+  ; -ᴹ‿involutive  to -ᴮ‿involutive
+  ; uniqueʳ‿-ᴹ     to uniqueʳ‿-ᴮ
+  ; ≈ᴹ-setoid      to ≈ᴮ-setoid
+  ; ≈ᴹ-sym         to symᴮ
+  ; leftSemimodule to leftSemimoduleᴮ
+  )
+open Properties leftSemimoduleᴮ
+  using () renaming
+  ( non-zeroʳ to non-zeroʳᴮ
+  ; non-zeroˡ to non-zeroˡᴮ
+  )
+open CommutativeRing ring
+  using (_≈_; _*_; _+_) renaming
+  ( Carrier to S
+  ; 0#      to 𝟘
+  ; 1#      to 𝟙
+  )
+open module Reasoningᴬ = Reasoning ≈ᴬ-setoid
```

Comment:
Needing to reason in 2 different setoids is also something that happens a lot. Rather than opening one of them globally (the one for `modB` in this case) and then renaming everything for `modA`, have you considered opening neither of them globally, but use the 'appropriate' one by using `where open ...` in the proofs themselves?

Reply:
Ah, right, I remember seeing that "trick" in some code and taking a mental note to use it.
Then, of course, I forgot. :(
Let me see where I can apply that, in my own code...

Reply:
**Changed included in latest push and ready for review.**